### DISCUSSION TITLE
Watching videos through Tor

### DISCUSSION THREAD

**Discussion by Kreyren** [FreeTubeApp/FreeTube#3285] [Q&A] [answered]
I am trying to replicate the following use:

/home/kreyren/.guix-profile/bin/torsocks /home/kreyren/.guix-profile/bin/mpv --ytdl-format="bestvideo[height=?720][fps<=?30][vcodec!=?vp9]+bestaudio/best" youtube-video-here

With the intention to watch youtube videos on a reasonable bitrate through mpv that uses tor due to ffmpeg

> Take note that SOCKS proxies aren't supported and https URLs also bypass the proxy. This is a limitation in FFmpeg. -- mpv irc channel

.. torsocks has to be used, but:

![torfreetube](https://user-images.githubusercontent.com/11302521/224467023-cd4b37e5-2b2f-4630-a23f-615cd40eaa68.png)

Does not launch the video and i dunno how to diagnose it -> help?

Proposed Solution: Manage the FFMPEG issue through implementing a way to use torsocks?

**Comment by absidue** [comment, marked as answer]
Hi @Kreyren

FreeTube calls the external executable directly without going through a shell, so quoting and escaping is not necessary and will instead result in the quotes and backslashes being passed into torsocks as part of the arguments. You also need to make it clear to FreeTube that `/home/kreyren/.guix-profile/bin/mpv` and `--ytdl-format=bestvideo[height=?720][fps<=?30][vcodec!=?vp9]+bestaudio/best` are two separate arguments, so that FreeTube knows to pass them to torsocks as two separate arguments, instead of one long one. You can do that by separating them with a `;` (click or hover on the question mark near the arguments box for that information).

With those changes, the contents of the custom external player arguments box should look like this:
`/home/kreyren/.guix-profile/bin/mpv;--ytdl-format=bestvideo[height=?720][fps<=?30][vcodec!=?vp9]+bestaudio/best`

Hope this helps.

**Reply by Kreyren** [reply]
Sorry i didn't get notification about it.

I tried the change:

![](https://i.imgur.com/g2Sjl2i.png)

But getting err and the external player is not working

**Reply by Kreyren** [reply]
My bad, it works i removed the local binary of torsocks so it can't find it, changing that to guix binary of torsocks works

Thanks for help!

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Kreyren did the answer solve your issue?

**Reply by Kreyren** [reply]
See https://github.com/FreeTubeApp/FreeTube/discussions/3285#discussioncomment-6037974
